core/src/ast_builder/expr/in_list.rs

```diff
@@ -19,60 +19,36 @@ impl From<Vec<ExprNode>> for InListNode {
     }
 }
 
-impl From<SelectNode> for InListNode {
-    fn from(node: SelectNode) -> Self {
-        let node = Box::new(node.into());
-        InListNode::Query(node)
-    }
-}
-
-impl From<GroupByNode> for InListNode {
-    fn from(node: GroupByNode) -> Self {
-        let node = Box::new(node.into());
-        InListNode::Query(node)
-    }
-}
-
-impl From<HavingNode> for InListNode {
-    fn from(node: HavingNode) -> Self {
-        let node = Box::new(node.into());
-        InListNode::Query(node)
-    }
-}
-
-impl From<LimitNode> for InListNode {
-    fn from(node: LimitNode) -> Self {
-        let node = Box::new(node.into());
-        InListNode::Query(node)
-    }
-}
-
-impl From<LimitOffsetNode> for InListNode {
-    fn from(node: LimitOffsetNode) -> Self {
-        let node = Box::new(node.into());
-        InListNode::Query(node)
+impl From<&str> for InListNode {
+    fn from(query: &str) -> Self {
+        InListNode::Text(query.to_owned())
     }
 }
 
-impl From<OffsetNode> for InListNode {
-    fn from(node: OffsetNode) -> Self {
-        let node = Box::new(node.into());
-        InListNode::Query(node)
+impl From<QueryNode> for InListNode {
+    fn from(node: QueryNode) -> Self {
+        InListNode::Query(Box::new(node))
```

Comment:
Same pattern works but `node: QueryNode` is already `QueryNode` so I thought that it would not be necessary to use `.into()`

Reply:
I don't think there is any extra cost. I'll look for a reference for him.

Reply:
> But.. for this case, it would be clear to use macro without QueryNode because QueryNode is not a part of select nodes but the others are all in the select modules. so the macro name is also impl_from_select_nodes

ok good